helpick-service/src/main/java/com/infoshare/service/NeedRequestService.java

```diff
@@ -26,27 +35,82 @@ public NeedRequestService(DB db) {
     public void createNeedRequest(String name, String location, String phone, TypeOfHelp typeOfHelp) {
         PersonInNeed personInNeed = new PersonInNeed(name, location, phone);
         NeedRequest needRequest = NeedRequest
-            .create(typeOfHelp, personInNeed);
+                .create(typeOfHelp, personInNeed);
         db.saveNeedRequest(needRequest);
     }
 
+    public void changeRequestStatus(List<NeedRequest> filteredList, int choice) {
```

Comment:
Tutaj mozna skorzystac z metody getRequestByUUID (tylko zmienic ja zeby NeedRequesta zamiast Optionala zwracala. I wydzielic metode do ustawiania statusu i zapisu danych do pliku. Wtedy kod sie mega skraca do: 
```suggestion
    ublic void changeRequestStatusToInProgress(UUID uuid) {
        updateNeedRequestStatusAndDateAndSave(getNeedRequestById(uuid), HelpStatuses.INPROGRESS);
    }
    
    private void updateNeedRequestStatusAndDateAndSave(NeedRequest changedRequest, HelpStatuses helpStatus) {
        changedRequest.setHelpStatus(helpStatus);
        changedRequest.setStatusChange(new Date());
         log.info("Status of request ID " + changedRequest.getUuid() + "restored to " + helpStatus.getStatus());
        db.saveNeedRequest(changedRequest);
    }
```

Reply:
Dzięki, ale zostawiam swoją wersję.

Reply:
wszystkie zmiany ok. W tym jednym przypadku powinien być zastosowany UUID w parametrze metody, ponieważ taki jest jego cel do jednoznacznej identyfikacji obiektu nad którym pracujemy. W tym wypadku przesyłanie List<NeedRequest> filteredList, int choice nie jest optymalne. Implementacja jest obojętna